docs/tutorials/http-serving-tutorial.md

```diff
@@ -30,43 +31,82 @@ static async insertGreeting(ctxt: TransactionContext<Knex>, friend: string, note
 DBOS provides endpoint decorators for all HTTP verbs used in APIs: [`@GetApi`](../api-reference/decorators#getapi), [`@PostApi`](../api-reference/decorators#postapi), [`@PutApi`](../api-reference/decorators.md#putapi), [`@PatchApi`](../api-reference/decorators.md#patchapi), and [`@DeleteApi`](../api-reference/decorators.md#deleteapi).
 Each associates a function with an HTTP URL.
 
-:::info
 
-You might be wondering why we don't talk about setting up an HTTP server.
-It's because DBOS is _serverless_.
-When you run an app locally with `npx dbos start`, we manage the HTTP server for you, using the endpoints and configuration you specify with decorators.
+### Inputs and HTTP Requests
 
-:::
+When a function has arguments other than its context, DBOS automatically parses them from the HTTP request, and returns an error to the client if they are not found.
 
-### Handlers
+Arguments can be parsed from three places:
 
-A function annotated with an endpoint decorator but no other decorators is called a _handler_ and must take a [`HandlerContext`](../api-reference/contexts#handlercontext) as its first argument, like in the first example above.
-Handlers can [invoke](../api-reference/contexts#handlerctxtinvoke) other functions and directly access HTTP requests and responses.
-However, DBOS makes no guarantees about handler execution: if a handler fails, it is not automatically retried.
-You should use handlers when you need to access HTTP requests or responses directly or when you are writing a lightweight task that does not need the strong guarantees of transactions and workflows.
+#### 1. URL Path Parameters
 
-### Inputs and HTTP Requests
+You can include a path parameter placeholder in a URL by prefixing it with a colon, like `name` in this example:
+
+```javascript
+@GetApi('/greeting/:name')
+static async greetingEndpoint(ctx: HandlerContext, name: string) {
+  return `Greeting, ${name}`;
+}
+```
+
+Then, give your method an argument with a matching name (such as `name: string` above) and it is automatically parsed from the path parameter.
+
+For example, if we send our app this request, then our method is called with `name` `dbos`:
+
+```
+GET /greeting/dbos
+```
+
+#### 2. URL Query String Parameters
+
+[`GET`](../api-reference/decorators#getapi) and [`DELETE`](../api-reference/decorators.md#deleteapi) endpoints automatically parse arguments from query strings.
```

Comment:
For other types of endpoints, you can use `@ArgSource(ArgSources.QUERY)` to explicitly parse an argument from query string.